### ISSUE TITLE
OpenFileGDB driver writing unicode values to fields

### ISSUE THREAD

**Issue by simonwright-jbarisk** [OSGeo/gdal#7503]
## Expected behavior and actual behavior.

Expected behaviour: String fields in a FileGDB created using the OpenFileGDB driver **don't** contain unicode codes such as:

`0.0m <= x <= 0.1m`.

Actual behaviour: String fields in a FileGDB created using the OpenFileGDB driver **contain** unicode codes such as:

`0.0m <= x <= 0.\udcc5\udcff`.

## Steps to reproduce the problem.

Pertinent Python code (stripped from a larger function):

```
units = 'm'

band_field = ogr.FieldDefn("DepthBand", ogr.OFTString)
layer.CreateField(band_field)

operator_str = '{min}{units} < x <= {max}{units}'

reclassargs = [[0.0, 0.1, 1.0], [0.1, 0.5, 2.0], [0.5, 2.0, 3.0], [2.0, 5.0, 4.0], [5.0, 10.0, 5.0]] # 

for feature in layer:

    feature.SetField("DepthBand", operator_str.format(min=band[0], max=band[1], units=units or ''))
   # Where band is selected from reclassargs based upon some logic

    layer.SetFeature(feature)

layer.SyncToDisk()
```

## Operating system

Rocky Linux version 8.7

## GDAL version and provenance

3.6.2 install into a Python 3.8.13 venv using pip install gdal=3.6.2

**Comment by rouault** [comment]
@simonwright-jbarisk Please provide a fully standalone reproducer, with input data if needed. For reference, I've tried the following script and it works fine:
```python
from osgeo import gdal, ogr
import os
import shutil

if os.path.exists("test_7503.gdb"):
    shutil.rmtree("test_7503.gdb")

ds = ogr.GetDriverByName("OpenFileGDB").CreateDataSource("test_7503.gdb")
layer = ds.CreateLayer("test", geom_type = ogr.wkbNone)
band_field = ogr.FieldDefn("DepthBand", ogr.OFTString)
layer.CreateField(band_field)
for i in range(10):
    f = ogr.Feature(layer.GetLayerDefn())
    layer.CreateFeature(f)
ds = None

ds = ogr.Open("test_7503.gdb", update = 1)
layer = ds.GetLayer(0)
for feature in layer:
    feature.SetField("DepthBand", "0.0m <= x <= 0.1m")
    layer.SetFeature(feature)
layer.SyncToDisk()
ds = None


ds = ogr.Open("test_7503.gdb")
layer = ds.GetLayer(0)
for feature in layer:
    assert feature["DepthBand"] == "0.0m <= x <= 0.1m"
```

**Comment by rouault** [comment]
@simonwright-jbarisk Given that you use SetFeature() and see corruptions, this might be just another effect of https://github.com/OSGeo/gdal/issues/7503, which will be fixed per https://github.com/OSGeo/gdal/pull/7525.
I'd be still be interested if you could come up with a reproducer to confirm this is the same issue

**Comment by simonwright-jbarisk** [comment]
Hi Evan

Managed to find time to create some test data and create a test script to demonstrate the issue.

The test.gdb is zipped to try and ensure that it transfers completely.

[test.gdb.zip](https://github.com/OSGeo/gdal/files/11110247/test.gdb.zip)

```
import os

from osgeo import gdal, ogr

print(f'gdal.__version__: {gdal.__version__}')


directory = '.'
file_name = 'test.gdb'
driver_name = 'OpenFileGDB'
layer_name = 'test'

driver = ogr.GetDriverByName(driver_name)
gdb = driver.Open(os.path.join(directory, file_name), 1)
layer = gdb.GetLayer(layer_name)

reclassargs = [[0.0, 0.1, 1.0],
               [0.1, 0.5, 2.0],
               [0.5, 2.0, 3.0],
               [2.0, 5.0, 4.0],
               [5.0, 10.0, 5.0]]

band_value_field_name = 'band_value'

depth_band_field_name = 'DepthBand'
depth_band_field = ogr.FieldDefn(depth_band_field_name, ogr.OFTString)
layer.CreateField(depth_band_field)

units = 'm'

for feature in layer:

    band_value = feature.GetField(band_value_field_name)
    band = None
    for i, arg in enumerate(reclassargs):
        if int(arg[2]) == int(band_value):
            if i == 0:
                operator_str = '{min}{units} <= x <= {max}{units}'
            elif i == len(reclassargs) - 1:
                operator_str = 'x > {min}{units}'
            else:
                operator_str = '{min}{units} < x <= {max}{units}'
            band = arg

    operator_str = operator_str.format(min=band[0], max=band[1], units=units)

    feature.SetField(depth_band_field_name, operator_str)

    layer.SetFeature(feature)

layer.SyncToDisk()

layer = None
gdb = None
```

The script opens the FileGDB in write mode, gets the layer, adds a new string field (DepthBand) to the layer, and, iterating over all the features in the layer, updates values in the new field based upon integer values in an existing field (band_value).

When I run the script some values in the new string field contain unicode characters.



Best wishes, Simon.

**Comment by rouault** [comment]
Running the above script, I confirm this is a duplicate of #7503, fixed per #7525

**Comment by simonwright-jbarisk** [comment]
Hi Evan

Many thanks for checking out my test data and script and confirming both the bug and the forthcoming fix.

Best wishes, Simon.
